Algo.h

```diff
@@ -294,7 +385,8 @@ class MMP
                 // invert wale pains???
 
                 for (auto &e : ((Vertex *)(source->p))->edges) {
-                    Interval ii(0, 0, 0, e->length(), 0, NULL, e);
+                    //check for invert
+                    Interval ii(0, 0, 0, e->length(), 0, NULL, e, false);
```

Comment:
why are you passing `invert` as false everywhere? You have to check whether the convention you are following on the current edge is right or wrong. Set `invert` accordingly. Check convention by seeing (in this case) whether the vertex is the on with the higher pointer value or not. If it is a lower pointer value, do not invert.